src/csharp/Microsoft.Spark/Sql/StorageLevel.cs

```diff
@@ -0,0 +1,131 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.Spark.Interop;
+using Microsoft.Spark.Interop.Ipc;
+
+namespace Microsoft.Spark.Sql
+{
+    /// <summary>
+    /// Flags for controlling the storage of an RDD. Each StorageLevel records whether to use
+    /// memory, whether to drop the RDD to disk if it falls out of memory, whether to keep the
+    /// data in memory in a JAVA-specific serialized format, and whether to replicate the RDD
+    /// partitions on multiple nodes. Also contains static properties for some commonly used
+    /// storage levels, MEMORY_ONLY.
+    /// </summary>
+    public sealed class StorageLevel : IJvmObjectReferenceProvider
+    {
+        private readonly JvmObjectReference _jvmObject;
+        private static readonly string s_storageLevelClassName =
+            "org.apache.spark.storage.StorageLevel";
+        private static StorageLevel s_diskOnly;
+        private static StorageLevel s_diskOnly2;
+        private static StorageLevel s_memoryOnly;
+        private static StorageLevel s_memoryOnly2;
+        private static StorageLevel s_memoryAndDisk;
+        private static StorageLevel s_memoryAndDisk2;
+        private static StorageLevel s_offHeap;
+
+        JvmObjectReference IJvmObjectReferenceProvider.Reference => _jvmObject;
+
+        internal StorageLevel(JvmObjectReference jvmObject)
+        {
+            UseDisk = (bool)jvmObject.Invoke("useDisk");
+            UseMemory = (bool)jvmObject.Invoke("useMemory");
+            UseOffHeap = (bool)jvmObject.Invoke("useOffHeap");
+            Deserialized = (bool)jvmObject.Invoke("deserialized");
+            Replication = (int)jvmObject.Invoke("replication");
+            _jvmObject = jvmObject;
+        }
+
+        public StorageLevel(
+            bool useDisk,
+            bool useMemory,
+            bool useOffHeap,
+            bool deserialized,
+            int replication = 1)
+        {
+            UseDisk = useDisk;
+            UseMemory = useMemory;
+            UseOffHeap = useOffHeap;
+            Deserialized = deserialized;
+            Replication = replication;
+            _jvmObject = SparkEnvironment.JvmBridge.CallConstructor(
+                s_storageLevelClassName,
+                UseDisk,
+                UseMemory,
+                UseOffHeap,
+                Deserialized,
+                Replication);
+        }
+
+        public bool UseDisk { get; private set; }
+        public bool UseMemory { get; private set; }
+        public bool UseOffHeap { get; private set; }
+        public bool Deserialized { get; private set; }
+        public int Replication { get; private set; }
+
+        public static StorageLevel DISK_ONLY
+        {
+            get => s_diskOnly ??= new StorageLevel(
+                (JvmObjectReference)SparkEnvironment.JvmBridge.CallStaticJavaMethod(
+                    s_storageLevelClassName,
+                    "DISK_ONLY"));
+        }
```

Comment:
Hmm good point, I guess we should do the same in that case. So our `MEMORY_ONLY` should be scala's `MEMORY_ONLY_SER`.

Reply:
Disregard ^ comment and change, Python serializes all it's records and so has no need to store in deserialized format in JVM ([related PR](https://github.com/apache/spark/commit/499ac3e69a102f9b10a1d7e14382fa191516f7b5#diff-fc5f6779209d9bdfa1ac1b49e62e3812)), but since C# creates Scala objects/dataframes, we should expose all the APIs as in Scala, hence making this change. Tested by persisting a `DataFrame` with deserialized `StorageLevel` and checked to see it's returning `StorageLevel` to be deserialized.

Reply:
Cool thanks for checking.